src/test/java/org/junitpioneer/jupiter/json/ArrayNodeToListTests.java

```diff
@@ -0,0 +1,216 @@
+				.executeTestMethodWithParameterTypes(ArrayNodeToListTests.class, "doesNotConvertNonNodesSimple",
+					String.class);
+
+		assertThat(results).hasNumberOfSucceededTests(2);
+		assertThat(results).hasNumberOfReportEntries(2).withValues("12", "23");
+	}
+
+	@Test
+	@DisplayName("does not use the converter for non-JSON arguments for List argument types")
+	void doesNotUseConverterList() {
+		ExecutionResults results = PioneerTestKit
+				.executeTestMethodWithParameterTypes(ArrayNodeToListTests.class, "doesNotConvertNonNodesList",
+					List.class);
+
+		assertThat(results).hasNumberOfSucceededTests(1);
+		assertThat(results).hasNumberOfReportEntries(1).withValues("[a, b, c]");
+	}
+
+	@Test
+	@DisplayName("throws a ParameterResolutionException if it can not convert complex objects")
+	void throwsForMalformedComplexObjects() {
+		ExecutionResults results = PioneerTestKit
+				.executeTestMethodWithParameterTypes(ArrayNodeToListTests.BadConfigurationTestCase.class,
+					"conversionException", List.class);
+
+		assertThat(results)
+				.hasSingleFailedContainer()
+				.withExceptionInstanceOf(UncheckedIOException.class)
+				.hasMessageContaining("Failed to convert to type");
+	}
+
+	@ParameterizedTest
+	@JsonClasspathSource(COMPOSERS)
+	@ReportEntry("{1}")
+	void convertFromClasspath(@Property("name") String name, @Property("music") List<String> works) {
+		then(name).isNotEmpty();
+		then(works).hasSize(2);
+	}
+
+	@ParameterizedTest
+	@JsonSource({ "{ 'single': [1, 4, 7] }", "{ 'single': [2, 4, 9] }" })
+	@ReportEntry("{0}")
+	void convertFromAnnotation(@Property("single") List<Integer> numbers) {
+		then(numbers).hasSize(3);
+	}
+
+	@ParameterizedTest
+	@JsonSource({ "{ 'statements': [true, false, true] }", "{ 'statements': [false, true, false] }" })
+	@ReportEntry("{0}")
+	void convertWithExplicitListType(@Property("statements") LinkedList<Boolean> numbers) {
+		then(numbers).hasSize(3);
+	}
+
+	@ParameterizedTest
+	@JsonClasspathSource(POETS)
+	@ReportEntry("{0}: {1}")
+	void convertToComplexObject(@Property("name") String name, @Property("works") List<Poem> poems) {
+		if (name.equals("T. S. Eliot")) {
+			then(poems).extracting(Poem::getRelease).noneMatch(i -> i < 1900);
+		} else {
+			then(poems).extracting(Poem::getRelease).noneMatch(i -> i > 1900);
+		}
+	}
+
+	@ParameterizedTest
+	@ValueSource(strings = { "12", "23" })
+	@ReportEntry("{0}")
+	void doesNotConvertNonNodesSimple(@Property("nope") String i) {
+		then(i).isNotNull();
+	}
+
+	@ParameterizedTest
+	@MethodSource("letters")
+	@ReportEntry("{0}")
+	void doesNotConvertNonNodesList(@Property("nope") List<String> letters) {
+		then(letters).hasSize(3).containsExactly("a", "b", "c");
+	}
+
+	public static Stream<Arguments> letters() {
+		return Stream.of(arguments(Arrays.asList("a", "b", "c")));
+	}
+
+	static class BadConfigurationTestCase {
+
+		@ParameterizedTest
+		@JsonClasspathSource(BAD_POEMS)
+		void conversionException(@Property("poems") List<Poem> poems) {
```

Comment:
Why does this test case not work? The only difference that I can spot between `bad_poems.json` and `poets.json` is the name of the property `poems` vs `works`. However, this shouldn't be causing a problem

Reply:
`release` in one case has "abcd" which is not a number.

Reply:
Got it, didn't saw that. Thanks